File: app/src/main/java/com/restart/spacestationtracker/data/SightSee.java
package com.restart.spacestationtracker.data;


import java.text.DecimalFormat;
import java.text.Format;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

/**
 * The type Sight see. Includes data to be able to identify an ISS flyby. Each flyby is represented
 * by a single SightSee.
 */
public class SightSee {
    /**
     * Variables to define a single SightSee. A list of SightSee all are represented by a single location.
     */
    private static final Format mSimpleDateFormat;
    private static final Format mDecimalFormat;
    private static String location;
    private final Date mRiseTimeDate;
    private final Date mSetTimeDate;
    private final String mDuration;
    private final String mRiseTime;

    /* Format all dates and decimals in the same manner. */
    static {
        mSimpleDateFormat = new SimpleDateFormat("MM/dd/yyyy hh:mm a", Locale.getDefault());
        mDecimalFormat = new DecimalFormat("#.#");
    }

    /**
     * Instantiates a new Sight see. Format any initialize any other variables.
     *
     * @param duration The duration the ISS will be approximately visible from rise to set.
     * @param risetime The risetime date time ISS will start to be visible.
     */
    public SightSee(int duration, int risetime) {
        mDuration = mDecimalFormat.format((double) duration / 60) + " minutes";

        mRiseTimeDate = new Date(risetime * 1000L);
        mRiseTime = mSimpleDateFormat.format(mRiseTimeDate);

        Calendar calendar = Calendar.getInstance(Locale.getDefault());
        calendar.setTime(mRiseTimeDate);
        calendar.add(Calendar.SECOND, duration);

        mSetTimeDate = new Date(calendar.getTimeInMillis());
    }

    /**
     * Gets duration.
     *
     * @return the duration
     */
    public String getDuration() {
        return mDuration;
    }

    /**
     * Gets rise time.
     *
     * @return the rise time
     */
    public String getRiseTime() {
        return mRiseTime;
    }

    /**
     * Gets location.
     *
     * @return the location
     */
    public static String getLocation() {
        return SightSee.location;
    }

    /**
     * Sets location.
     *
     * @param location the location
     */
    public static void setLocation(String location) {
        SightSee.location = location;
    }

    /**
     * Gets rise time date.
     *
     * @return the rise time date
     */
    public Date getRiseTimeDate() {
        return mRiseTimeDate;
    }

    /**
     * Gets set time date.
     *
     * @return the set time date
     */
    public Date getSetTimeDate() {
        return mSetTimeDate;
    }
}

File: app/src/main/java/com/restart/spacestationtracker/data/SightSee.kt
package com.restart.spacestationtracker.data

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class SightSee(
    val startUTC: Long,
    val endUTC: Long,
    val durationSeconds: Int,
    val magnitude: Double,
    val maxElevation: Double,
    val startAzimuthCompass: String,
    val endAzimuthCompass: String,
    val location: String,
) {
    val riseTimeDate: Date by lazy { Date(startUTC * 1000L) }
    val setTimeDate: Date by lazy { Date(endUTC * 1000L) }

    val formattedRiseTime: String by lazy {
        SimpleDateFormat("h:mm a", Locale.getDefault()).format(riseTimeDate)
    }

    val formattedDuration: String by lazy {
        val minutes = durationSeconds / 60
        val seconds = durationSeconds % 60
        "$minutes min $seconds sec"
    }

    val formattedDateHeader: String by lazy {
        SimpleDateFormat("EEEE, MMMM d", Locale.getDefault()).format(riseTimeDate).uppercase(Locale.getDefault())
    }

    companion object {
        var location: String = ""
    }
}